airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -224,6 +224,49 @@ definitions:
       $parameters:
         type: object
         additionalProperties: true
+  SelectiveAuthenticator:
+    title: Selective Authenticator
+    description: Authenticator that selects concrete authenticator based on config property.
+    type: object
+    additionalProperties: true
+    required:
+      - type
+      - authenticators
+      - authenticator_selection_path
+    properties:
+      type:
+        type: string
+        enum: [SelectiveAuthenticator]
+      authenticator_selection_path:
+        title: Authenticator Selection Path
+        description: Path of the field in config with selected authenticator name
+        type: array
+        items:
+          type: string
+        examples:
+          - ["auth"]
+          - ["auth", "type"]
+      authenticators:
+        title: Authenticators
+        description: Authenticators to select from.
+        type: object
+        additionalProperties:
+          anyOf:
```

Comment:
can you define this anyOf in #/definitions/Authenticator to avoid the need to repeat the whole union twice?

Reply:
@girarda I was thinking about this, but one union has SelectiveAuthenticator and another don't

Reply:
I'm fine with the selective authenticator being self-referencing. I don't have a real use case in mind, but an authenticator is an authenticator

Reply:
```
============= 31 failed, 2459 passed, 1031 warnings in 45.69s ==============
```
🙄 I will see if this is something that can be easy to fix, but will timebox it to few hours

Reply:
🤷 such is life. not a big deal to be honest

Reply:
ok, looks like this will not compatible with our model_to_component factories. datamodel-generator will create additional type for each `$ref`. This creates problem with create_component, _create_component_from_model etc